packages/headless/src/api/knowledge/knowledge-api.ts

```diff
@@ -0,0 +1,44 @@
+import {
+  BaseQueryFn,
+  createApi,
+  FetchArgs,
+  fetchBaseQuery,
+  FetchBaseQueryError,
+  retry,
+} from '@reduxjs/toolkit/query';
+import {ConfigurationSection} from '../../state/state-sections';
+
+type StateNeededByKnowledgeAPI = ConfigurationSection;
+
+/**
+ * `dynamicBaseQuery` is passed to the baseQuery of the createApi,
+ * but note that the baseQuery will not be used if a queryFn is provided in the createApi endpoint
+ */
+const dynamicBaseQuery: BaseQueryFn<
+  string | FetchArgs,
+  unknown,
+  FetchBaseQueryError
+> = async (args, api, extraOptions) => {
+  const state = api.getState() as StateNeededByKnowledgeAPI;
+  const updatedArgs = {
+    ...(args as FetchArgs),
+    headers: {
+      ...((args as FetchArgs)?.headers || {}),
+      Authorization: `Bearer ${state.configuration.accessToken}`,
+    },
+  };
+  try {
+    const data = fetchBaseQuery({
+      baseUrl: `${state.configuration.platformUrl}/rest/organizations/${state.configuration.organizationId}`,
+    })(updatedArgs, api, extraOptions);
+    return {data};
+  } catch (error) {
+    return {error: error as FetchBaseQueryError};
+  }
+};
+
+export const knowledgeApi = createApi({
```

Comment:
should be answer api

Reply:
I do not think so. I defeat the purpose of reusing a single slice to handle many apis.
If down the road we have many apis to handle under the knowledge use case, the `answerApi` will only be one of them.

KnowledgeApi is only the name of the React Toolkit Query Wrapper. The real endoint calling the answerApi is declared [here](https://github.com/coveo/ui-kit/pull/4014/files#diff-c08f41924a6308395c6062336daabdfbd4717d924f8439b03d64b6f1eda80388R112)

The container here is used to handle the base query. And the common shenanigans that we could want to make generic. Its pretty barebone right now.

In the future, it can be a possibility to have this slice just called `api` without the need to `knowledge` at all. And use a single instance to control all the coveo APIs used by headless